tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -213,8 +213,14 @@ def _fetch_resources(self, query):
             return self.augment(self.source.get_resources(query)) or []
         except HttpError as e:
             error = extract_error(e)
-            if error is None:
+            error_code = extract_error_code(e)
+            error_message = extract_error_message(e)
```

Comment:
one thought here is rather than decoding the response twice from json, that we just unify the extraction to a single method to return message and code, though that relies on updating extant callers of extract which afaicr is just this location and tests.

Reply:
That makes sense. I couldn't find any uses of the `extract_error` method besides this location so I joined all the extract error methods into one, renamed it, and edited it return all 3 values we want from the error.